components/speedreader/speedreader_throttle.h

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_SPEEDREADER_SPEEDREADER_THROTTLE_H_
+#define BRAVE_COMPONENTS_SPEEDREADER_SPEEDREADER_THROTTLE_H_
+
+#include "base/memory/weak_ptr.h"
+#include "services/network/public/mojom/url_response_head.mojom-forward.h"
+#include "third_party/blink/public/common/loader/url_loader_throttle.h"
+
+namespace speedreader {
+
+// TODO(iefremov): ad-hoc tool for initial testing. Combines the hardcoded
+// array and the ';'-separated list from command-line.
+bool IsWhitelisted(const GURL& url);
+
+// Launches the speedreader distillation pass over a reponce body, deferring
+// the load until distillation is done.
+// TODO(iefremov): Avoid distilling the same page twice (see comments in
```

Comment:
this could be a little tricky because url alone is not sufficient. Maybe etag or other cache headers when available? Do we have any way to know if this was served from cache or not because that would be the most reliable method imo

Reply:
This TODO is mostly about an internal `URLLoader` pipeline... Checking how distillation works with caching and or etags is interesting indeed